AnalysisTools/python/templates/AddMetFilters.py

```diff
@@ -0,0 +1,34 @@
+from UWVV.AnalysisTools.AnalysisFlowBase import AnalysisFlowBase
+from UWVV.Utilities.helpers import parseChannels
+import FWCore.ParameterSet.Config as cms
+
+# From https://twiki.cern.ch/twiki/bin/viewauth/CMS/MissingETOptionalFiltersRun2
+class AddMetFilters(AnalysisFlowBase):
+    def __init__(self, *args, **kwargs):
+        super(AddMetFilters, self).__init__(*args, **kwargs)
+
+    def makeAnalysisStep(self, stepName, **inputs):
+        step = super(AddMetFilters, self).makeAnalysisStep(stepName, **inputs)
+        
+        if stepName == 'initialStateEmbedding':
+            from RecoMET.METFilters.BadPFMuonFilter_cfi import BadPFMuonFilter 
+            BadPFMuonFilter.muons = step.getObjTag('m') 
+            BadPFMuonFilter.PFCandidates = cms.InputTag("packedPFCandidates")
+            step.addModule("BadPFMuonFilter", BadPFMuonFilter)
+            
+            from RecoMET.METFilters.BadChargedCandidateFilter_cfi import BadChargedCandidateFilter
+            BadChargedCandidateFilter.muons = step.getObjTag('m') 
+            BadChargedCandidateFilter.PFCandidates = cms.InputTag("packedPFCandidates")
+            step.addModule("BadChargedCandidateFilter", BadChargedCandidateFilter)
+
+            channels = inputs.pop('channels', [])
```

Comment:
Did this get changed?

Reply:
Ah no I forgot this one. But I don't think I really understand the issue. Channels is the name of the final states to be looped over since the bad muon filters (its really just this and has nothing to do with MET, I should change the name) get embedded into the final state. Is there another way to get this information? What would you suggest I change the name to?